packages/stable/src/__tests__/api/datapoints.int.spec.ts

```diff
@@ -25,9 +55,20 @@ describe('Datapoints integration test', () => {
   beforeAll(async () => {
     client = setupLoggedInClient();
     [timeserie] = await client.timeseries.create([{ name: 'tmp' }]);
+    await client.spaces.upsert([testSpace]);
+    await client.instances.upsert({
+      items: [timeseriesCdmInstance],
+    });
   });
   afterAll(async () => {
     await client.timeseries.delete([{ id: timeserie.id }]);
+    await client.instances.delete([
```

Comment:
One downside here is that a failure/exception will cause the instances to not be cleaned up in the CDF project.
We added some clean up here: https://github.com/cognitedata/cognite-sdk-js/pull/1083/files#diff-8bd79ffb443e5a0606897f789e00907501054c1bd8bd81a6295edaa5625d0cedR51

Reply:
I think afterAll will only not run after tests if there was some unexpected failure on jest side? (I guess also if the timeseries delete throws) so not sure how important a clean up is right now (I think maybe we can see if this becomes a problem and then introduce a cleanup?)

Also just a comment that in DM tests we try to avoid deleting/recreating spaces and containers because it bloats the elastic index, so we for example do not clean up spaces after tests on purpose. Instances however should be cleaned up I agree. :slightly_smiling_face: